src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -728,26 +727,34 @@ internal void DownloadAndInstall(PackageDownloadHandle packageDownloadHandle, st
                                     MessageBoxButton.OK, MessageBoxImage.Error);
                             }
                         }
-
-                        if (packageDownloadHandle.Extract(DynamoViewModel.Model, downloadPath, out dynPkg))
-                        {
-                            pmExtension.PackageLoader.LoadPackages(new List<Package> {dynPkg});
-                            packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Installed;
-                        }
-                        else
-                        {
-                            packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Error;
-                            packageDownloadHandle.Error(Resources.MessageInvalidPackage);
-                        }
+                        InstallStateCheck(packageDownloadHandle, downloadPath);
                     }
                     catch (Exception e)
                     {
                         packageDownloadHandle.Error(e.Message);
                     }
                 }));
-
             });
+        }
 
+        /// <summary>
+        /// Check Dynamo package install state
+        /// </summary>
+        /// <param name="packageDownloadHandle">package download handle</param>
+        /// <param name="downloadPath">package download path</param>
+        internal void InstallStateCheck(PackageDownloadHandle packageDownloadHandle, string downloadPath)
```

Comment:
Minor thing: As we are assigning the package state in this method, may be we can call it something like SetPackageState.

Reply:
@reddyashish Make sense. Let me update!